client/finality-grandpa/src/until_imported.rs

```diff
@@ -77,6 +80,63 @@ pub(crate) enum DiscardWaitOrReady<Block: BlockT, W, R> {
 	Ready(R),
 }
 
+/// Prometheus metrics for the `UntilImported` queue.
+//
+// At a given point in time there can be more than one `UntilImported` queue. One can not register a
+// metric twice, thus queues need to share the same Prometheus metrics instead of instantiating
+// their own ones.
+//
+// When a queue is dropped it might still contain messages. In order for those to not distort the
+// Prometheus metrics, the `Metric` struct cleans up after itself within its `Drop` implementation
+// by subtracting the local_waiting_messages (the amount of messages left in the queue about to
+// be dropped) from the global_waiting_messages gauge.
+pub(crate) struct Metrics {
+	global_waiting_messages: Gauge<U64>,
+	local_waiting_messages: u64,
+}
+
+impl Metrics {
+	pub(crate) fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+		Ok(Self {
+			global_waiting_messages: register(Gauge::new(
+				"finality_grandpa_until_imported_waiting_messages_number",
+				"Number of waiting finality grandpa messages within the until imported queue.",
```

Comment:
```suggestion
				"Number of finality grandpa messages waiting within the until imported queue.",
```

This just sounds better imo, and rolls off the tongue nicely because of the alliteration.